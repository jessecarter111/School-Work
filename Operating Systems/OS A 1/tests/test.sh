#!/bin/sh

PREFIX=test
TESTS='0 1 2 3 4 5 6 7 8 9 10 11 12 13 14'
PROG=runme.sh
count=0

if [ ! -x $PROG ]; then
  echo Error: $PROG is not in the current directory or is not executable
  exit 1
fi

for T in $TESTS; do
  if [ -f $PREFIX.$T.xo ]; then  
    echo ===========================================
    echo Test file: $PREFIX.$T.xo 
    ./$PROG 100000 $PREFIX.$T.xo > $PREFIX.$T.out
    if diff $PREFIX.$T.out $PREFIX.$T.gold > /dev/null; then
      echo " " PASSED
      let "count = count + 1"
    else
      echo " " FAILED
    fi
    # rm $PREFIX.$T.out
  else
    echo Oops: file $PREFIX.$T.xo does not exist!
  fi
done

echo =============================================
echo $count tests passed
