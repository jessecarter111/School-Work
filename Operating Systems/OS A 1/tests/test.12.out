00> PC: 0002, State: 0002
00> Registers: 
00>      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 1, r0

00> PC: 0006, State: 0002
00> Registers: 
00>      r00:0001 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 65535, r0

00> PC: 000a, State: 0002
00> Registers: 
00>      r00:ffff r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 0, r0

00> PC: 000e, State: 0002
00> Registers: 
00>      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 10, r0

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:000a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 65526, r0

00> PC: 0016, State: 0002
00> Registers: 
00>      r00:fff6 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 256, r0

00> PC: 001a, State: 0002
00> Registers: 
00>      r00:0100 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 1, r0

00> PC: 001e, State: 0002
00> Registers: 
00>      r00:0001 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 65535, r0

00> PC: 0022, State: 0002
00> Registers: 
00>      r00:ffff r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 1, r0

00> PC: 0026, State: 0002
00> Registers: 
00>      r00:0001 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 65535, r0

00> PC: 002a, State: 0002
00> Registers: 
00>      r00:ffff r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 0, r0

00> PC: 002e, State: 0002
00> Registers: 
00>      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 256, r0

00> PC: 0032, State: 0002
00> Registers: 
00>      r00:0100 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: (null) 

